Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -633,59 +656,17 @@ extension RouteController: Router {
     }
     
     public func userIsOnRoute(_ location: CLLocation, status: NavigationStatus?) -> Bool {
-        
-        guard let destination = routeProgress.currentLeg.destination else {
-            preconditionFailure("Route legs used for navigation must have destinations")
-        }
-        
-        // If the user has arrived, do not continue monitor reroutes, step progress, etc
-        if routeProgress.currentLegProgress.userHasArrivedAtWaypoint &&
-            (delegate?.router(self, shouldPreventReroutesWhenArrivingAt: destination) ??
-                DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint) {
-            return true
-        }
-        
-        // If we still wait for the first status from NavNative, there is no need to reroute
-        guard let status = status ?? sharedNavigator.mostRecentNavigationStatus else { return true }
-
-        /// NavNative doesn't support reroutes after arrival.
```

Comment:
I take it MapboxNavigationNative has already implemented rerouting after arrival? If not, we would need to document this regression as a known issue in the release notes.

Reply:
Could someone from @mapbox/navnative please describe how reroutes after arrival work?

Reply:
In the end, we'v decided to keep this logic on platform side, so I restored original code.